string name
uint8 type
---